### PR TITLE
Fix #13222 : Vehicle Collision causes negative number of passengers

### LINKED ISSUE #13222: Vehicle collission causing negative number of passengers
**OS:** Linux
**Version:** 0.3.1
**Commit/Build:** 645289f4ec1b1c5ed1475da63190153d99f0b406

After a collision in which everyone died (28), info panel for the ride listed -28 passengers on ride. Save/reload fixed issue.

- [ ] Reproducible in RCT2 (vanilla)?
- [ ] Specific to multiplayer?

**Steps to reproduce:**
1. Have fatal crash of all trains on a ride

**Screenshots / Video:**
loopy-lenny
negative-customers

**Save game:**
[ll.zip](https://github.com/OpenRCT2/OpenRCT2/files/5395024/ll.zip)

### PR DESCRIPTION
**Pull request by raacker**
when peep_sprite_remove is called, RemoveFromRide() will decrease ride's num_riders

### CODE REVIEW

**Comment by duncanspumpkin** [comment]
I think this will need a changelog entry. I'm pretty sure it's an original bug from my look at the assembly.

**Comment by duncanspumpkin** [comment]
Also will need network increment.

**Comment by raacker** [comment]
@duncanspumpkin Could you tell me what do you mean by "network increment"?

**Comment by duncanspumpkin** [comment]
Network.cpp has a version number just increment the number by one. It prevents folk from getting desyncs due to different simulation states as they won't be able to connect with different network versions

**Comment by tupaschoal** [comment]
I think it is `NetworkBase.cpp` now, but I'm not 100% sure

**Comment by raacker** [comment]
@duncanspumpkin I see thanks. Just updated network version number.
@tupaschoal And it was correct. NetworkBase.cpp has NETWORK_STREAM_VERSION

@Broxzier Hey I also added this fix to changelog because it has been there for a while even before 3.1.0. Is it correct then?

**Comment by Broxzier** [comment]
> @Broxzier Hey I also added this fix to changelog because it has been there for a while even before 3.1.0. Is it correct then?

Yep!

File: distribution/changelog.txt [hunk 1]
```diff
@@ -33,6 +33,7 @@
 - Fix: [#13129] Missing error message when waiting for train to leave station on the ride measurements graph.
 - Fix: [#13138] Fix logical sorting of list windows.
 - Fix: [#13158] Cursors are drawn incorrectly in text input fields.
+- Fix: [#13222]  Vehicle Collision causes negative number of passengers.
```
Comment (AaronVanGeffen) [inline distribution/changelog.txt:36]: Stray space:
```suggestion
- Fix: [#13222] Vehicle Collision causes negative number of passengers.
```

Comment (raacker) [reply distribution/changelog.txt]: Thanks for the good catch :)

File: distribution/changelog.txt [hunk 2]
```diff
@@ -34,6 +34,7 @@
 - Fix: [#13129] Missing error message when waiting for train to leave station on the ride measurements graph.
 - Fix: [#13138] Fix logical sorting of list windows.
 - Fix: [#13158] Cursors are drawn incorrectly in text input fields.
+- Fix: [#13222] Vehicle Collision causes negative number of passengers.
```
Comment (Gymnasiast) [inline distribution/changelog.txt:37]:
```suggestion
- Fix: [#13222] Vehicle collision causes negative number of passengers.
```

File: distribution/changelog.txt [hunk 3]
```diff
@@ -34,6 +34,7 @@
 - Fix: [#13129] Missing error message when waiting for train to leave station on the ride measurements graph.
 - Fix: [#13138] Fix logical sorting of list windows.
 - Fix: [#13158] Cursors are drawn incorrectly in text input fields.
+- Fix: [#13222] Vehicle collision causes negative number of passengers.
```
Comment (Gymnasiast) [inline distribution/changelog.txt:37]: Aw, shite, should have added ` (original bug)`. :/

**Pull request merged**